### ISSUE TITLE
cannot run app on Android 4.4 device

### ISSUE THREAD

**Issue by rogerwq** [facebook/react-native#2885]
## I am trying with React Native Android for the awesome project. It was successful running under emulator while fail with the real device with android 4.4. Here below is the output:

```
...
Installed on 1 device.

BUILD SUCCESSFUL

Total time: 10.449 secs

This build could be faster, please consider using the Gradle Daemon: http://gradle.org/docs/2.4/userguide/gradle_daemon.html
Starting the app (adb shell am start -n com.awesomeproject/.MainActivity)...
Starting: Intent { cmp=com.awesomeproject/.MainActivity }
Error type 3
Error: Activity class {com.awesomeproject/com.awesomeproject.MainActivity} does not exist.
```

***Can anyone give a hint?***

**Comment by arypurnomoz** [comment]
I also got this problem

**Comment by NishanthShankar** [comment]
The documentation provided is for Android 5.0 and above as `adb reverse` works only from Android 5.0 and above. 
I got a "Unable to download JS bundle" error on the device when I tried to run it on a KitKat device
I solved it by building the bundle file using the command 

`curl "http://localhost:8081/index.android.bundle?platform=android" -o "android/app/src/main/assets/index.android.bundle"`

 followed by `react-native run-android`
(Create the necessary folders if they don't exist)
The only problem being that I have build and run every time I make a change

**Comment by rogerwq** [comment]
@NishanthShankar The problem I met is the application is not started (using USB Debugging mode). I  encountered "Unable to download JS bundle" error when I played with the emulator. And it was solved by "brew update watchman". Worth to have a try to avoid manual building every time.

**Comment by NishanthShankar** [comment]
@rogerwq . Yeah I had to update watchman to get it running on the emulator too. But that doesn't seem to work for the device. I guess a fix for the same is in the pipeline. I'm not sure why the app fails to launch.

**Comment by oren** [comment]
is this related to my issue? https://github.com/facebook/react-native/issues/2871

**Comment by kmagiera** [comment]
@rogerwq By any chance do you have more than one device connected (e.g. real device + emulator). What is the output of `adb devices` when you get the error you've posted?

Also for running on device <5.0 please refer to this section: http://facebook.github.io/react-native/docs/running-on-device-android.html#configure-device-to-connect-to-the-dev-server-via-wi-fi

**Comment by rogerwq** [comment]
@kmagiera there is only 1 real device.

```
List of devices attached 
phone-id	device
```

The "wifi" approach seems impossible for me. I have no *rage shake menu* since there is no app launched in real device. Even the console says *Installed on 1 device*, I cannot find any icon of the installed app.

Any further suggestion?

**Comment by kmagiera** [comment]
1. Do you have USB debugging enabled on your device: http://facebook.github.io/react-native/docs/running-on-device-android.html#usb-debugging
2. Try building app manually using gradle: `cd android` and `./gradlew :app:installDebug`
3. If the above fails try running adb command to install apk (should provide more output) from `android` dir in your project: `adb install app/build/outputs/apk/app-debug.apk`

**Comment by oren** [comment]
@kmagiera thanks for the suggestions. Both option 1 and 2 installed the app on my phone but I still see the red screen with "Unable to download JS bundle".

**Comment by rogerwq** [comment]
@kmagiera Great thanks for the support. 

I got the white screen with "Welcome to React Native" finally by building, installing manually  the apk file and connecting to dev server via wifi.

**Comment by journeys** [comment]
The same issue caused by the short of system storage.
```
adb install app/build/outputs/apk/app-debug.apk
```
will get the error msg, but gradle install will not.

**Comment by sidaphextwin** [comment]
End adb.exe process and run react-native android-run again. This solutions is good in my computer.

**Comment by kingnand** [comment]
@NishanthShankar you save me a lot. Thanks

**Comment by tiagofumo** [comment]
@NishanthShankar You can set the development server `ip:port` on the app_menu>"Dev Settings">"Debug server host & port for device" in order to reload via wifi.

**Comment by ababba15** [comment]
Also guys check if you have nice permissions, that was my problem.

**Comment by alejandromcsd** [comment]
I was facing this issue after installing-uninstalling a release version of the RN app on my device. I was no longer able to do a yarn android.

I solved it with @kmagiera suggested solution (#3: from `android` dir in your project: `adb install app/build/outputs/apk/app-debug.apk`). Thanks

**Comment by fawaz-ahmed** [comment]
In my case had to do what @NishanthShankar has suggested and also some additional steps:

1_ create dir `mkdir /android/app/src/main/assets`
2_ `curl "http://localhost:8081/index.android.bundle?platform=android" -o "android/app/src/main/assets/index.android.bundle"`
3_ create file index.android.js `touch index.android.js`

//sample index.android.js 
import { AppRegistry } from 'react-native';
import App from './app/index';
AppRegistry.registerComponent('<app name here>', () => App);
